### PR TITLE
added reportporta testresult collector.

### PR DESCRIPTION
**Pull request by gopiprasanthpotipireddy**
the first version of reportportal testresults collector without unit tests.

### CODE REVIEW

**Comment by gopiprasanthpotipireddy** [comment]
@rvema can you review it now.

File: application.properties [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+		# MongoDB replicaset
+		#dbreplicaset=[false if you are not using MongoDB replicaset]
+		#dbhostport=[host1:port1,host2:port2,host3:port3]
+
+		# Database Username - default is blank
+		dbusername=dashboarduser
+
+		# Database Password - default is blank
+		dbpassword=dbpassword
```
Comment (rvema) [inline application.properties]: please refrain from storing username and password in the property files in github. Please point to the documentation mentioning that this needs to be updated.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+  <url>https://github.com/capitalone/Hygieia</url>
+  <packaging>jar</packaging>
+
+  <artifactId>reportportal_test_results_collector</artifactId>
+
+  <parent>
+    <groupId>com.capitalone.dashboard</groupId>
+    <artifactId>Hygieia</artifactId>
+    <version>3.0.2-SNAPSHOT</version>
+    <relativePath>../../..</relativePath>
```
Comment (rvema) [inline pom.xml]: please remove this dependency on relativePath. Every collector should be on its own.

File: src/main/java/com/capitalone/dashboard/collector/DefaultReportPortalClient.java [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+             report.setTestId(collectorItemId);
+             report.setExecutionId(launchId);
+             report.setCollectorItemId(project.getId());
+             report.setTestCapabilities(caps);
+             report.setTotalCount(count);
+             report.setDuration((((Double) project.getOptions().get("approximateDuration")).longValue()));
+             report.setStartTime((long)(project.getOptions().get("start_time")));
+             report.setEndTime((long)(project.getOptions().get("end_time")));
+             reports.add(report);
+                //project.setInstanceUrl(instanceUrl);
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/DefaultReportPortalClient.java]: please remove the commented code.

File: src/main/java/com/capitalone/dashboard/collector/DefaultReportPortalClient.java [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+     try {
+
+         for (Object obj : parseAsArray(step_url,"content")) {
+             JSONObject testData = (JSONObject) obj;
+             TestCaseStep step1=new TestCaseStep();
+     		step1.setDescription(str(testData,"name"));
+     		step1.setStatus(getStatus(str(testData,"status")));
+     		JSONObject stats=(JSONObject) testData.get("statistics");
+     		JSONObject executions=(JSONObject) stats.get("executions");
+     		//step.setFailedTestStepCount((integer(executions,"failed")));
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/DefaultReportPortalClient.java]: please remove the commented code.

File: src/main/java/com/capitalone/dashboard/collector/ReportPortalCollectorTask.java [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+
+    @Override
+    public void collect(ReportPortalCollector collector) {
+        long start = System.currentTimeMillis();
+
+        Set<ObjectId> udId = new HashSet<>();
+        udId.add(collector.getId());
+        List<ReportPortalProject> existingProjects = reportProjectRepository.findByCollectorIdIn(udId);
+        List<ReportPortalProject> latestProjects = new ArrayList<>();
+        //clean(collector, existingProjects);
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/ReportPortalCollectorTask.java]: please remove the commented lines of code.

File: src/main/java/com/capitalone/dashboard/collector/ReportPortalSettings.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+@ConfigurationProperties(prefix = "reportportal")
+public class ReportPortalSettings {
+    private String cron;
+    private String username;
+    private String password;
+    private List<String> servers;
+    private String bearerToken;
+    private List<String> niceNames;
+    private String projectName;
+	//private String instanceUrl;
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/ReportPortalSettings.java]: please remove the commented line of code

File: src/main/java/com/capitalone/dashboard/collector/RestOperationsSupplier.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestOperations;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class RestOperationsSupplier implements Supplier<RestOperations> {
+    public RestOperations get() {
+        HttpComponentsClientHttpRequestFactory requestFactory = new HttpComponentsClientHttpRequestFactory();
+        requestFactory.setConnectTimeout(20000);
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/collector/RestOperationsSupplier.java:13]: please have the timeout parameters configurable through the properties

File: src/main/java/com/capitalone/dashboard/model/ReportPortalCollector.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+            protoType.getReportPortalServers().addAll(servers);
+        }
+       
+        
+        if (!CollectionUtils.isEmpty(niceNames)) {
+            protoType.getNiceNames().addAll(niceNames);
+        }
+
+        Map<String, Object> allOptions = new HashMap<>();
+        //allOptions.put(ReportPortalProject.INSTANCE_URL,"" );
```
Comment (rvema) [inline src/main/java/com/capitalone/dashboard/model/ReportPortalCollector.java]: please remove the commented lines of code

**Review by rvema — CHANGES_REQUESTED** [review]
please look into the review comments and address them. Do ping me on gitter in case you need additional information

**Review by gopiprasanthpotipireddy — COMMENTED** [review]
@rvema updated the changes. can you look into it now.

**Review by rvema — APPROVED** [review]
lgtm

**Pull request merged**